### PR TITLE
IconColorizedLayered: For layered and colored icons

### PR DESCRIPTION
**Pull request by keenfoong**
to allow for both colored icons and default colored icons. Icons can also be layered on top of each other then merged for interesting effects. May replace icon() and iconColorized(). 

Still needs clean up and documentation. Will revisit.

### CODE REVIEW

File: zoo/libs/iconlib/iconlib.py [hunk 1]
```diff
@@ -156,6 +156,82 @@ def iconColorized(cls, iconName, size=16, color=(255, 255, 255), overlayName=Non
...
+        :param colors: 3 tuple for the icon color
+        :type colors: list of tuple
+        :return: the colorized icon
+        :param overlayName: The name of the icon that will be overlayed on top of the original icon
+        :param overlayColor: The colour of the overlay
+        :rtype: QtGui.QIcon
+        """
+        defaultSize = 1
+
+        if isinstance(iconNames, basestring):
```
Comment (dsparrow27) [inline zoo/libs/iconlib/iconlib.py]: don't type check , iconNames should only be one type... a list.

File: zoo/libs/iconlib/iconlib.py [hunk 2]
```diff
@@ -156,6 +156,82 @@ def iconColorized(cls, iconName, size=16, color=(255, 255, 255), overlayName=Non
...
+        if isinstance(iconScaling, list):
+            iconScaling = list(iconScaling)  # copy
+
+        if not isinstance(colors, list):
+            colors = [colors]
+        else:
+            colors = list(colors)  # copy
+
+        if iconNames is []:
+            print "WARNING: iconNames cannot be none for iconColorizedLayered"
```
[the remark this answers is not in the document]
Comment (dsparrow27) [reply zoo/libs/iconlib/iconlib.py]: no prints and don't compare via `is []`
instead just do 
`if not iconNames:`

File: zoo/libs/iconlib/iconlib.py [hunk 3]
```diff
@@ -156,6 +156,82 @@ def iconColorized(cls, iconName, size=16, color=(255, 255, 255), overlayName=Non
...
+        :param overlayName: The name of the icon that will be overlayed on top of the original icon
+        :param overlayColor: The colour of the overlay
+        :rtype: QtGui.QIcon
+        """
+        defaultSize = 1
+
+        if isinstance(iconNames, basestring):
+            iconNames = [iconNames]
+        elif isinstance(iconNames, list):
+            iconNames = list(iconNames)  # copy
```
Comment (dsparrow27) [inline zoo/libs/iconlib/iconlib.py]: why copy? I don't see any point in this function where in place modification is done

File: zoo/libs/iconlib/iconlib.py [hunk 4]
```diff
@@ -156,6 +156,82 @@ def iconColorized(cls, iconName, size=16, color=(255, 255, 255), overlayName=Non
...
+
+        if isinstance(iconScaling, list):
+            iconScaling = list(iconScaling)  # copy
+
+        if not isinstance(colors, list):
```
Comment (dsparrow27) [inline zoo/libs/iconlib/iconlib.py]: why typecheck?, we should expect a list and nothing else.

File: zoo/libs/iconlib/iconlib.py [hunk 5]
```diff
@@ -206,11 +283,45 @@ def addOverlay(cls, pixmap, overlayPixmap, color):
...
+        if color is not None:
+            overlayPixmap = cls.colorPixmap(overlayPixmap, color)
 
         # Paint the overlay pixmap over the original
         painter = QtGui.QPainter(pixmap)
         painter.setCompositionMode(QtGui.QPainter.CompositionMode_SourceOver)
+
+        x = y = 0
+        if align is QtCore.Qt.AlignCenter:
+            x = pixmap.width() / 2 - overlayPixmap.width() / 2
```
Comment (dsparrow27) [inline zoo/libs/iconlib/iconlib.py]: use `*0.5` instead of divide

File: zoo/libs/iconlib/iconlib.py (hunk 1 above)
Comment (keenfoong) [reply zoo/libs/iconlib/iconlib.py]: Cool yeah, I had a feeling this might not be too good. What should be done instead here? Separate methods?

**Pull request merged**